miner/worker.go

```diff
@@ -551,32 +529,27 @@ func (w *worker) taskLoop() {
 func (w *worker) resultLoop() {
 	for {
 		select {
-		case block := <-w.resultCh:
-			// Short circuit when receiving empty result.
-			if block == nil {
+		case result := <-w.resultCh:
+			// Short circuit when receiving empty result
+			if result == nil {
 				continue
 			}
+			block := result.SealedBlock
 			// Short circuit when receiving duplicate result caused by resubmitting.
 			if w.chain.HasBlock(block.Hash(), block.NumberU64()) {
 				continue
 			}
+
 			var (
 				sealhash = w.engine.SealHash(block.Header())
 				hash     = block.Hash()
 			)
-			w.pendingMu.RLock()
-			task, exist := w.pendingTasks[sealhash]
-			w.pendingMu.RUnlock()
-			if !exist {
```

Comment:
Please add a check (along with the error message) that is similar to this, where your check will make sure the `result.State` and `result.Receipts` are not nil.   If they are nil, then there is something very wrong (most likely a bug in the code).

Reply:
`result.Receipts` could be `nil` if a block has no txns.
`result.State` cannot be nil, will add a nil check for `result.State`